src/OneExplorer.ts

```diff
@@ -0,0 +1,183 @@
+  ) {
+    super(label, collapsibleState);
+
+    this.tooltip = `${this.label}`;
+
+    if (node.type === NodeType.config) {
+      this.iconPath = new vscode.ThemeIcon('gear');
+    }
+    else if (node.type === NodeType.directory){
+      this.iconPath = vscode.ThemeIcon.Folder;
+    }
+    else if (node.type === NodeType.model) {
+      this.iconPath = vscode.ThemeIcon.File;
+    }
+  }
+}
+
+export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
+  private _onDidChangeTreeData: vscode.EventEmitter<OneNode|undefined|void> =
+      new vscode.EventEmitter<OneNode|undefined|void>();
+  readonly onDidChangeTreeData: vscode.Event<OneNode|undefined|void> =
+      this._onDidChangeTreeData.event;
+
+  private cfgMap: Node|undefined;
+
+  constructor(private workspaceRoot: vscode.Uri|undefined) {
+    if (workspaceRoot !== undefined) {
+      this.cfgMap = this.getConfigMap(workspaceRoot);
+    }
+  }
+
+  // TODO(dayo): enable refresh command
+  refresh(): void {
+    this._onDidChangeTreeData.fire();
+  }
+
+  getTreeItem(element: OneNode): vscode.TreeItem {
+    element.command = { command: 'oneExplorer.openFile', title: "Open File", arguments: [element.node] };
+    return element;
+  }
+
+  getChildren(element?: OneNode): OneNode[]|Thenable<OneNode[]> {
+    if (this.cfgMap === undefined) {
+      vscode.window.showInformationMessage('No ONE model or config in empty workspace');
+      return Promise.resolve([]);
+    }
+
+    if (element) {
+      return Promise.resolve(this.getNode(element.node));
+    } else {
+      return Promise.resolve(this.getNode(this.cfgMap));
+    }
+  }
+
+  private getNode(node: Node): OneNode[] {
+    const toOneNode = (node: Node): OneNode => {
+      if (node.childs.length > 0) {
+        return new OneNode(node.name, vscode.TreeItemCollapsibleState.Collapsed, node);
+      } else {
+        return new OneNode(node.name, vscode.TreeItemCollapsibleState.None, node);
+      }
+    };
+
+    return node.childs.map(node => toOneNode(node));
+  }
+
+  private getConfigMap(rootPath: vscode.Uri): Node {
+    const node = {type: NodeType.directory, name: path.parse(rootPath.fsPath).base, dir: true, childs: [], uri: rootPath};
+    this.searchConfigs(node, path.dirname(rootPath.fsPath));
+    return node;
+  }
+
+  private searchConfigs(node: Node, dirPath: string) {
+    const dirpath = path.join(dirPath, node.name);
+    const files = fs.readdirSync(dirpath);
+
+    for (const fn of files) {
+      const fpath = path.join(dirpath, fn);
+      const fstat = fs.statSync(fpath);
+
+      if (fstat.isDirectory()) {
+        const dirNode = {type: NodeType.directory, name: fn, dir: true, childs: [], uri: vscode.Uri.file(fpath)};
+
+        this.searchConfigs(dirNode, dirpath);
+        if (dirNode.childs.length > 0) {
+          node.childs.push(dirNode);
+        }
+      }
+      else if ((fstat.isFile() 
+        && (fn.endsWith('.tflite')) || fn.endsWith('.onnx'))){
```

Comment:
How about supporting the tensorflow model ?

_Originally posted by @jyoungyun in https://github.com/Samsung/ONE-vscode/pull/495#discussion_r858163175_

Reply:
Done